encoder.go

```diff
@@ -293,47 +290,40 @@ type analyzeEncFunc func(byte) bool
 // case2: could not use EncModeNumeric, but you can find all of them in character mapping, use EncModeAlphanumeric.
 // case3: could not use EncModeAlphanumeric, but you can find all of them in ISO-8859-1 character set, use EncModeByte.
 // case4: could not use EncModeByte, use EncModeJP, no more choice.
-//
 func analyzeEncodeModeFromRaw(raw []byte) encMode {
-	analyzeFnMapping := map[encMode]analyzeEncFunc{
-		EncModeNumeric:      analyzeNum,
-		EncModeAlphanumeric: analyzeAlphaNum,
-		EncModeByte:         nil,
-		EncModeJP:           nil,
-	}
-
-	var (
-		f    analyzeEncFunc
-		mode = EncModeNumeric
-	)
-
-	// loop to check each character in raw data,
-	// from low mode to higher while current mode could bearing the input data.
+	mode := EncModeNumeric
 	for _, byt := range raw {
-	reAnalyze:
-		if f = analyzeFnMapping[mode]; f == nil {
-			break
+		// Start from most inclusive and work backwards
+		// Check for each kind starting with most common then expand out to least inclusive.
+		if analyzeNum(rune(byt)) {
```

Comment:
Is there any meaning of converting a byte to a rune? raw is a byte slice

Reply:
Characters are runes in go. Using rune instead of byte encourages comparing to a single character which is the point of these functions since they aren’t Unicode aware.